pkg/reconciler/knativeserving/ingress/kourier.go

```diff
@@ -107,6 +108,28 @@ func configureGWServiceType(instance *v1beta1.KnativeServing) mf.Transformer {
 	}
 }
 
+// configureGWServiceLoadBalancerIP configures Kourier GW's service loadBalancerIP.
+func configureGWServiceLoadBalancerIP(instance *v1beta1.KnativeServing) mf.Transformer {
```

Comment:
As per https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/
> Only applies to Service Type: LoadBalancer.

The `ServiceLoadBalancerIP` applies to service type `LoadBalancer` only and you also need to use it for `LoadBalancer`, right? Then, I would the add `if` condition in `configureGWServiceType()` to insert the logic to apply `ServiceLoadBalancerIP` like:

```go
                        case v1.ServiceTypeClusterIP, v1.ServiceTypeNodePort, v1.ServiceTypeLoadBalancer:
                                svc.Spec.Type = serviceType
                                if serviceType == v1.ServiceTypeNodePort && instance.Spec.Ingress.Kourier.ServiceLoadBalancerIP != ""{
                                        ...
                                }
```

Probably we should rename the function `configureGWServiceType()`  with `configureGWService()` then.

Reply:
It's not that simple. The default type for the `kourier` Service is `LoadBalancer`
https://github.com/knative/operator/blob/a3758e7915b60ea859cbc4951101708d92097302/cmd/operator/kodata/ingress/1.10/kourier/kourier.yaml#L576

If we do this, we'd have to change this early return check because a user might set `service-loadBalancerIP` but not `service-type`.
https://github.com/knative/operator/blob/037957ad5d3ccd22b12826a0a17f22b122d78a32/pkg/reconciler/knativeserving/ingress/kourier.go#L82-L84

I ended up refactoring the whole function. Let me know if you prefer this or if I should revert.